File: find_pool.py
import urllib.request, json

url = 'https://yields.llama.fi/pools'
req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
try:
    with urllib.request.urlopen(req) as response:
        print("Downloaded. Parsing JSON...")
        data = json.loads(response.read().decode())
        print("Filtering pools...")
        pools = [p for p in data['data'] if ('WBTC' in p['symbol'] or 'BTCB' in p['symbol']) and p.get('apyBaseBorrow', 0) > 0 and p.get('tvlUsd', 0) > 10000000]
        pools.sort(key=lambda x: x['tvlUsd'], reverse=True)
        for p in pools[:5]:
            print(f"{p['project']} ({p['chain']}) - {p['symbol']}")
            print(f"Borrow: {p.get('apyBaseBorrow')} | Supply: {p.get('apyBase')} | TVL: ${p.get('tvlUsd')/1e6:.1f}M")
            print(f"ID: {p['pool']}\n---")
except Exception as e:
    print(f"Error: {e}")

File: find_pool.js
const https = require('https');

https.get('https://yields.llama.fi/pools', (res) => {
    let body = '';
    res.on('data', d => body += d);
    res.on('end', () => {
        try {
            const data = JSON.parse(body);
            const pools = data.data.filter(p =>
                (p.symbol.includes('WBTC') || p.symbol.includes('BTCB')) &&
                p.apyBaseBorrow > 0 &&
                p.tvlUsd > 10000000
            ).sort((a, b) => b.tvlUsd - a.tvlUsd).slice(0, 5);

            pools.forEach(p => {
                console.log(`${p.project} (${p.chain}) - ${p.symbol}`);
                console.log(`Borrow APY: ${p.apyBaseBorrow} | Supply APY: ${p.apyBase} | TVL: ${p.tvlUsd}`);
                console.log(`ID: ${p.pool}`);
                console.log('---');
            });
        } catch (e) {
            console.error('Error parsing JSON:', e);
        }
    });
}).on('error', e => console.error(e));
